core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -1019,7 +1019,11 @@ class SnappySession(@transient private val sc: SparkContext,
               userSpecifiedSchema = userSpecifiedSchema,
               partitionColumns = partitionColumns,
               options = params).write(mode, data)
-            (r, Some(r.schema))
+            if (None != userSpecifiedSchema) {
+              (r, Some(userSpecifiedSchema.get))
+            } else {
+              (r, Some(r.schema))
+            }
```

Comment:
@SachinJanani Right, so invoke createRelation separately as being done in JdbcExtendedUtils.externalResolvedDataSource